swap_meet/vendor.py

```diff
@@ -1,2 +1,58 @@
 class Vendor:
-    pass
\ No newline at end of file
+    
+    def __init__(self, inventory=None):
+        inventory = inventory if inventory is not None else []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        items_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        return items_by_category
+    
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+        my_index = self.inventory.index(my_item)
+        their_index = other.inventory.index(their_item)
+        self.inventory[my_index] , other.inventory[their_index] = other.inventory[their_index] , self.inventory[my_index]
+        return True
+
+
+    def swap_first_item(self, other):
+        if not self.inventory or not other.inventory:
+            return False
+        self.inventory[0], other.inventory[0] = other.inventory[0] , self.inventory[0]
+        return True
+
+    
+    def get_best_by_category(self, category):
+        condition = -1
+        best_item = None
+        for item in self.inventory:
+            if item.category == category and item.condition > condition:
+                best_item = item
+                condition = item.condition
+        return best_item
```

Comment:
This is a great approach to find the best item in a single pass over the data. If we don't need to be concerned about adding another iteration over the input, we could reuse some code by pairing the `get_by_category` function and `max` to get the result:
```suggestion
        items_in_category = self.get_by_category(category)
        return max(items_in_category, lambda item: item.condition)
```